gpytorch/lazy/constant_mul_lazy_tensor.py

```diff
@@ -55,50 +55,50 @@ class ConstantMulLazyTensor(LazyTensor):
         >>> #   [ 2, 1.5, 1 ]]]
     """
 
-    def __init__(self, lazy_var, constant):
+    def __init__(self, base_lazy_tensor, constant):
         if torch.is_tensor(constant):
             if constant.ndimension() > 1:
                 raise RuntimeError(
                     "Got a constant with %d dimensions - expected a 0D or 1D tensor" % constant.ndimension()
                 )
             elif constant.numel() > 1:
-                if not (lazy_var.ndimension() == 3 and lazy_var.size(0) == constant.numel()):
+                if not (base_lazy_tensor.ndimension() == 3 and base_lazy_tensor.size(0) == constant.numel()):
                     numel = constant.numel()
                     raise RuntimeError(
                         "A constant with size %d expedts a 3D lazy var. with batch size %d. "
                         "Got a %dD lazy var. with size %s"
-                        % (numel, numel, lazy_var.ndimension(), repr(lazy_var.size()))
+                        % (numel, numel, base_lazy_tensor.ndimension(), repr(base_lazy_tensor.size()))
                     )
 
             elif constant.numel() == 1:
                 constant = constant.squeeze()
         else:
-            constant = torch.tensor(constant, device=lazy_var.device, dtype=torch.float32)
+            constant = torch.tensor(constant, device=base_lazy_tensor.device, dtype=torch.float32)
```

Comment:
seems like this should be using the dtype of the base_lazy_tensor as well

Reply:
Updated